### PR TITLE
Handles TILEDB_CHAR array metadata

### PR DESCRIPTION
**Pull request by antalakas**

### CODE REVIEW

File: array.go [hunk 1]
```diff
@@ -1386,7 +1404,7 @@ func getMetadataValue(datatype Datatype, valueNum uint, cvalue unsafe.Pointer) (
 			value = *(*float64)(unsafe.Pointer(cvalue))
 		}
 	case TILEDB_CHAR:
-		tmpslice := (*[1 << 46]C.char)(unsafe.Pointer(cvalue))[:valueNum:valueNum]
+		tmpslice := (*[1 << 30]C.char)(unsafe.Pointer(cvalue))[:valueNum:valueNum]
```
Comment (Shelnutt2) [inline array.go:1407]: Is there a reason we can't use 46 for the size of the array here and are limited to 32 bit size?

Comment (antalakas) [reply array.go:1407]: It does using 46 works too here and it seems that i still cannot reproduce the error

Comment (antalakas) [reply array.go:1407]: It seems that the only way to have a segfault here is for `cvalue` to be `nil`

Comment (antalakas) [reply array.go]: Checked with vcf-sample-20/data array and reproduced the error. Indeed for index 4 `cvalue` is `null` pointer, `valueNum=1`. Can this be a bug in core?

Comment (Shelnutt2) [reply array.go]: What is the metadata key for the value that is null?

Comment (antalakas) [reply array.go]: Screenshot 2020-07-14 at 12 35 57 AM

**Pull request merged**